packages/gatsby-remark-code-repls/src/index.js

```diff
@@ -134,12 +134,10 @@ module.exports = (
             "package.json": {
               content: {
                 dependencies: dependencies.reduce((map, dependency) => {
-                  if (dependency.includes(`@`)) {
-                    const [name, version] = dependency.split(`@`)
-                    map[name] = version
-                  } else {
-                    map[dependency] = `latest`
-                  }
+                  const [, name, , version] = /^(@?.+?)(@(.+))?$/.exec(
```

Comment:
Do you think you could find a cleaner or more self-documenting solution here?

A bit of a nit, but this is really hard to follow.

Reply:
What about:
```javascript
if (dependency.includes(`@`)) {
  const splitIndex = dependency.findLastIndex('@')
  const name = dependency.substr(0, splitIndex)
  const version = dependency.substr(splitIndex + 1)
  map[name] = version
} else {
  map[dependency] = `latest`
}
```
If you are fine with this version, do you still want a unit test for it? If so, I think its better to extract it to function to a separate file and make some test for it. Anyways, I think after adding unit test I should also cover most complicated cases, where we have two or more @ signs and at least one of them is not at position 0. That test will force me to complicate the code. But I'm leaving this decision up to you.

Reply:
@krzysztofpniak yeah, I'm more aligned with that version, but that also doesn't really work in its current form.

Given:

```js
const getNameAndVersion = (dependency, map) => {
  if (dependency.includes(`@`)) {
    const splitIndex = dependency.findLastIndex('@')
    const name = dependency.substr(0, splitIndex)
    const version = dependency.substr(splitIndex + 1)
    map[name] = version
  } else {
    map[dependency] = `latest`
  }
  return map
}
```

`getNameAndVersion('react')` `{ react: 'latest' }`
`getNameAndVersion('react@16.0.0')` // dependency.findLastIndex is not a function at getNameAndVersion

I'm definitely in favor of breaking out and adding a few unit tests to assert functionality here.

Reply:
Sorry for the delay. Now it is ok, I also added unit tests.

Reply:
I know it's late - but maybe this would be most self documenting approach - https://www.npmjs.com/package/npm-package-arg ;)

Reply:
@pieh using external libraries for such a small things increases node_modules size. We live in the days when people create libraries for everything but IMHO this is not the way. Of course if library is widely used in project it makes sense. In this situation this is just one call of one simple function.

Reply:
@krzysztofpniak we are sympathetic to that--but using those small little packages also tends to have a lot of other benefits, e.g.

- We don't need to test the code
- We get any fixes (or new features, updates, etc.) for free just by using the package

So for Node.js code--we really recommend using packages if they are available and can be used.

Reply:
Would you be up for making that change? I'd recommend using that library--great idea @pieh

Reply:
Of course. Done.